core/types/block_ext_test.go

```diff
@@ -106,3 +111,410 @@ func assertDifferentPointers[T any](t *testing.T, a *T, b any) {
 	// Note: no need to check `b` is of the same type as `a`, otherwise
 	// the memory address would be different as well.
 }
+
+func TestBlockExtraGetWith(t *testing.T) {
+	t.Parallel()
+
+	b := &Block{}
+
+	extra := extras.Block.Get(b)
+	require.NotNil(t, extra)
+	assert.Equal(t, &BlockBodyExtra{}, extra)
+
+	extra = &BlockBodyExtra{
+		Version: 1,
+	}
+	extras.Block.Set(b, extra)
+
+	extra = extras.Block.Get(b)
+	wantExtra := &BlockBodyExtra{
+		Version: 1,
+	}
+	assert.Equal(t, wantExtra, extra)
+}
+
+func TestBodyExtraGetWith(t *testing.T) {
+	t.Parallel()
+
+	b := &Body{}
+
+	extra := extras.Body.Get(b)
+	require.NotNil(t, extra)
+	assert.Equal(t, &BlockBodyExtra{}, extra)
+
+	extra = &BlockBodyExtra{
+		Version: 1,
+	}
+	extras.Body.Set(b, extra)
+
+	extra = extras.Body.Get(b)
+	wantExtra := &BlockBodyExtra{
+		Version: 1,
+	}
+	assert.Equal(t, wantExtra, extra)
+}
+
+func TestBodyExtraRLP(t *testing.T) {
```

Comment:
This and the `Block` equivalent both need `coreth` golden data. An RLP round trip is only part of the story because the _exact_ encoding format is necessary.

Reply:
Added for both body and block (and checked in master branch)